functions/src/models/offers/index.ts

```diff
@@ -30,17 +33,21 @@ export class Offer implements IOffer {
     pinUserRef: DocumentReference<DocumentData>,
     requestRef: DocumentReference<DocumentData>,
     cavUserSnapshot: User,
+    requestSnapshot: Request | null,
     message: string,
     status: OfferStatus,
     createdAt = Timestamp.now(),
+    seenAt = Timestamp.now(),
```

Comment:
You shouldn't be assigning the default value of `seenAt` as now, because then every offer created would have the value of `seenAt` to be the moment it's created.
instead it should be set to `null` in the constructor

Reply:
+1

Reply:
Fixed, I've set it as Timestamp.now() as default value and my reasoning for it was because essentially when an offer is created or updated, an activity is happening at that current moment to generate a notification

But setting the seenAt time in the setOffer action does the same thing and makes more sense
`if (!(payload instanceof Offer)) {
    const offerPayload = Offer.factory({
      ...payload,
      seenAt: firestore.Timestamp.now(),
    });
    dispatch({
      type: SET,
      payload: {
        offerPayload,
        offerId,
      },
      firebase: offerId ? setUserOffer : createUserOffer,
    });
 } else {
    // eslint-disable-next-line no-param-reassign
    payload.seenAt = firestore.Timestamp.now();
    dispatch({
      type: SET,
      payload: {
        payload,
        offerId,
      },
      firebase: offerId ? setUserOffer : createUserOffer,
    });
}
`